futures/tests/ready_queue.rs

```diff
@@ -55,13 +55,19 @@ fn resolving_errors() {
 
         drop(tx2);
 
-        assert_eq!(Poll::Ready(Some(Err(oneshot::Canceled))), queue.poll_next_unpin(cx));
```

Comment:
I think the bad thing about this PR is that we cannot do this.

Reply:
However, actually, this is not just a problem for this PR. I also encountered this in #1785.